arangod/Sharding/ShardingInfo.cpp

```diff
@@ -128,25 +139,27 @@ ShardingInfo::ShardingInfo(arangodb::velocypack::Slice info, LogicalCollection*
     }
   }
 
-  auto minReplicationFactorSlice = info.get(StaticStrings::MinReplicationFactor);
-  if (!minReplicationFactorSlice.isNone()) {
-    if (minReplicationFactorSlice.isNumber()) {
-      _minReplicationFactor = minReplicationFactorSlice.getNumber<size_t>();
-      if (!isSatellite() && _minReplicationFactor > _replicationFactor) {
+  if (!isASatellite) {
+    auto writeConcernSlice = info.get(StaticStrings::MinReplicationFactor);
+    if (!writeConcernSlice.isNone()) {
+      if (writeConcernSlice.isNumber()) {
+        _writeConcern = writeConcernSlice.getNumber<size_t>();
+        if (!isSatellite() && _writeConcern > _replicationFactor) {
+          THROW_ARANGO_EXCEPTION_MESSAGE(
+              TRI_ERROR_BAD_PARAMETER,
+              "writeConcern cannot be larger than replicationFactor (" +
```

Comment:
Do we want to name it writeConcern here?
as far as I see this is only the "internal" name and we have to stick to minRepicationFactor over API, so I guess user might be confused?